django_mongodb_backend/lookups.py

```diff
@@ -56,34 +56,15 @@ def inner(self, compiler, connection):
 def get_subquery_wrapping_pipeline(self, compiler, connection, field_name, expr):  # noqa: ARG001
     return [
         {
-            "$facet": {
-                "group": [
-                    {
-                        "$group": {
-                            "_id": None,
-                            "tmp_name": {
-                                "$addToSet": expr.as_mql(compiler, connection, as_expr=True)
-                            },
-                        }
-                    }
-                ]
-            }
-        },
-        {
-            "$project": {
-                field_name: {
-                    "$ifNull": [
-                        {
-                            "$getField": {
-                                "input": {"$arrayElemAt": ["$group", 0]},
-                                "field": "tmp_name",
-                            }
-                        },
-                        [],
-                    ]
-                }
+            "$group": {
+                "_id": None,
+                # use a temporal name in order to support field_name="_id"
+                "tmp_name": {"$addToSet": expr.as_mql(compiler, connection, as_expr=True)},
             }
         },
+        {"$unionWith": {"pipeline": [{"$documents": [{"tmp_name": []}]}]}},
```

Comment:
I guess this won't work with QE (but maybe the query isn't supported for some other reason).

A problem with our current testing setup is that we have to replicate all the tests we want to do with QE under the "encryption_" test app.

So if we have two different code paths in this method and under `if self.wrap_for_global_aggregation:` in `query.py`, we won't have very comprehensive testing.

Something to think about.